utils/directories.cc

```diff
@@ -128,6 +128,9 @@ future<> directories::verify_owner_and_mode(fs::path path) {
         if (!can_access) {
             verification_error(std::move(path), "Directory cannot be accessed for read, write, and execute");
         }
+        if (level && !recurse) {
+            co_return;
+        }
         co_await lister::scan_dir(path, {}, [] (fs::path dir, directory_entry de) -> future<> {
             co_await verify_owner_and_mode(dir / de.name);
```

Comment:
Shouldn't it call `do_...` version with `level++`ed?

Reply:
Or wait, we don't need `level` at all -- just don't dive into recursion if `!recurse`

Reply:
Ah, it's called for plain files too, so probably we do need level. In this case it should call the `do_...` with increased level as I suggested above

Reply:
hmm, that's odd.  I had it this way locally and I probably didn't amend the patch before pushing....